src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyCommentFilter.java

```diff
@@ -392,29 +392,69 @@ public int hashCode() {
          * @return true if the source of event matches the text of this tag.
          */
         public boolean isMatch(TreeWalkerAuditEvent event) {
-            final int line = event.getLine();
             boolean match = false;
-
-            if (line >= firstLine && line <= lastLine) {
-                final Matcher tagMatcher = tagCheckRegexp.matcher(event.getSourceName());
-
-                if (tagMatcher.find()) {
-                    match = true;
-                }
-                else if (tagMessageRegexp == null) {
-                    if (event.getModuleId() != null) {
-                        final Matcher idMatcher = tagCheckRegexp.matcher(event.getModuleId());
-                        match = idMatcher.find();
-                    }
+            if (isInScopeOfSuppression(event)) {
+                if (isCheckMatch(event)) {
+                    match = isMessageMatch(event);
                 }
-                else {
-                    final Matcher messageMatcher = tagMessageRegexp.matcher(event.getMessage());
-                    match = messageMatcher.find();
+                else if (isIdMatch(event)) {
+                    match = isMessageMatch(event);
                 }
             }
             return match;
         }
 
+        /**
+         * Checks whether the {@link TreeWalkerAuditEvent} is in the scope of the suppression.
+         * @param event {@link TreeWalkerAuditEvent} instance.
+         * @return true if the {@link TreeWalkerAuditEvent} is in the scope of the suppression.
+         */
+        private boolean isInScopeOfSuppression(TreeWalkerAuditEvent event) {
+            final int line = event.getLine();
+            return line >= firstLine && line <= lastLine;
+        }
+
+        /**
+         * Checks whether {@link TreeWalkerAuditEvent} source name matches the check format.
+         * @param event {@link TreeWalkerAuditEvent} instance.
+         * @return true if the {@link TreeWalkerAuditEvent} source name matches the check format.
+         */
+        private boolean isCheckMatch(TreeWalkerAuditEvent event) {
+            final Matcher checkMatcher = tagCheckRegexp.matcher(event.getSourceName());
+            return checkMatcher.find();
+        }
+
+        /**
+         * Checks whether the {@link TreeWalkerAuditEvent} module ID matches the ID format.
+         * @param event {@link TreeWalkerAuditEvent} instance.
+         * @return true if the {@link TreeWalkerAuditEvent} module ID matches the ID format.
+         */
+        private boolean isIdMatch(TreeWalkerAuditEvent event) {
+            boolean match = false;
+            if (event.getModuleId() != null) {
+                final Matcher idMatcher = tagCheckRegexp.matcher(event.getModuleId());
```

Comment:
in a future PR, this could instead match using a `tagIdRegexp` instead of reusing `tagCheckRegexp`